addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -306,7 +306,9 @@ const Wysiwyg = Widget.extend({
                 $el.selectElement();
 
                 if (!$el.parent().hasClass('o_stars')) {
-                    self.openMediaDialog(params);
+                    // Waiting for all the options to be initialized before
+                    // opening the media dialog.
+                    self.snippetsMenu._mutex.getUnlockedDef().then(() => self.openMediaDialog(params));
```

Comment:
It might be worth checking if the node still exists before calling the media dialog.  Though this is unlikely going to happen, you could technically ask the mutex to delete a node containing an image, then double click on it. Now it will wait for the image to be deleted, but it will still invoke the media dialog for the deleted image.